store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreProperties.java

```diff
@@ -0,0 +1,130 @@
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreConstants.KEY_SKIP_FAILED_FEDERATED_STORE_EXECUTE;
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreConstants.PROPERTIES;
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreConstants.SCHEMA;
+
+public class FederatedStoreProperties extends StoreProperties {
+
+    public FederatedStoreProperties() {
+        setStoreClass(FederatedStore.class);
+        setStorePropertiesClass(FederatedStoreProperties.class);
+    }
+
+    public void setGraphIds(final String graphIdsCSV) {
+        set(KEY_GRAPH_IDS, graphIdsCSV);
+    }
+
+    public void setTrueSkipFailedExecution() {
+        setSkipFailedExecution(true);
+    }
+
+    public void setFalseSkipFailedExecution() {
+        setSkipFailedExecution(false);
+    }
+
+    public void setCustomPropertyAuths(final String auths) {
+        set(getCustomPropsKey(), auths);
+    }
+
+    public static String getCustomPropsKey() {
+        return String.format("%s.%s", GAFFER_FEDERATED_STORE, CUSTOM_PROPERTIES_AUTHS);
+    }
+
+    public void setGraphAuth(final String graphId, final String authCSV) {
+        set(getGraphAuthsKey(graphId), authCSV);
+    }
+
+    public void setSkipFailedExecution(final boolean b) {
+        set(KEY_SKIP_FAILED_FEDERATED_STORE_EXECUTE, Boolean.toString(b));
+    }
+
+    public void setGraphPropFile(final String graphId, final String file) {
+        final String key = getGraphConfigKey(graphId, GraphConfigEnum.properties, LocationEnum.file);
+        set(key, file);
+    }
+
+    public void setGraphSchemaFile(final String graphId, final String file) {
+        final String key = getGraphConfigKey(graphId, GraphConfigEnum.schema, LocationEnum.file);
+        set(key, file);
+    }
+
+    public void setGraphPropId(final String graphId, final String file) {
+        final String key = getGraphConfigKey(graphId, GraphConfigEnum.properties, LocationEnum.id);
+        set(key, file);
+    }
+
+    public void setGraphSchemaId(final String graphId, final String file) {
+        final String key = getGraphConfigKey(graphId, GraphConfigEnum.schema, LocationEnum.id);
+        set(key, file);
+    }
+
+    public static String getGraphConfigKey(final String graphId, final GraphConfigEnum graphConfigEnum, final LocationEnum locationEnum) {
+        return String.format("%s.%s.%s.%s", GAFFER_FEDERATED_STORE, graphId, graphConfigEnum.value, locationEnum.value);
+    }
+
+
+    public static String getGraphAuthsKey(final String graphId) {
+        return String.format("%s.%s.%s", GAFFER_FEDERATED_STORE, graphId, AUTHS);
+    }
+
+    public enum GraphConfigEnum {
+        schema(SCHEMA), properties(PROPERTIES);
+
+        private final String value;
+
+        GraphConfigEnum(final String value) {
+            this.value = value;
+        }
+    }
+
+    public enum LocationEnum {
+        file(FILE), id(FederatedStoreConstants.ID);
```

Comment:
Please can you use upper case for enum constants.

Reply:
👍